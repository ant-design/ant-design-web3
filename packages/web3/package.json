{
  "name": "@ant-design/web3",
  "version": "1.17.2",
  "type": "module",
  "main": "dist/esm/index.js",
  "module": "dist/esm/index.js",
  "typings": "dist/esm/index.d.ts",
  "exports": {
    "import": "./dist/esm/index.js",
    "require": "./dist/lib/index.js",
    "types": "./dist/esm/index.d.ts"
  },
  "sideEffects": false,
  "files": [
    "dist",
    "CHANGELOG.md",
    "README.md"
  ],
  "keywords": [
    "ant",
    "component",
    "components",
    "design",
    "framework",
    "frontend",
    "react",
    "react-component",
    "ui",
    "web3"
  ],
  "homepage": "https://web3.ant.design",
  "bugs": {
    "url": "https://github.com/ant-design/ant-design-web3/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ant-design/ant-design-web3"
  },
  "scripts": {
    "dev": "father dev",
    "build": "father build"
  },
  "dependencies": {
    "@ant-design/cssinjs": "^1.21.1",
    "@ant-design/icons": "^5.3.7",
    "@ant-design/web3-assets": "workspace:*",
    "@ant-design/web3-common": "workspace:*",
    "@ant-design/web3-icons": "workspace:*",
    "@ctrl/tinycolor": "^4.1.0",
    "@inline-svg-unique-id/react": "^1.2.3",
    "antd": "^5.21.5",
    "bs58": "^6.0.0",
    "classnames": "^2.5.1",
    "copy-to-clipboard": "^3.3.3",
    "decimal.js": "^10.4.3",
    "lodash-es": "^4.17.21"
  },
  "devDependencies": {
    "@ant-design/web3-bitcoin": "workspace:*",
    "@ant-design/web3-eth-web3js": "workspace:*",
    "@ant-design/web3-ethers": "workspace:*",
    "@ant-design/web3-solana": "workspace:*",
    "@ant-design/web3-sui": "workspace:*",
    "@ant-design/web3-wagmi": "workspace:*",
    "@ant-design/web3-ton": "workspace:*",
    "@types/react": "^18.3.5",
    "@types/react-dom": "^18.3.0",
    "father": "^4.4.4",
    "typescript": "^5.6.2",
    "@types/lodash-es": "^4.17.12"
  },
  "peerDependencies": {
    "antd": ">=5.0.0",
    "react": ">=17.0.0",
    "react-dom": ">=17.0.0"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  }
}
