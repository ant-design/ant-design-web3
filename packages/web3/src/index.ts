export * from './connector';
export * from './connect-button';
export * from './connect-modal';
export * from './address';
export * from './browser-link';
export * from './nft-image';
export * from './nft-card';
export * from './crypto-price';
export * from './crypto-input';
export * from './token-select';
export * from './hooks';
export * from '@ant-design/web3-common';
// export Web3ConfigProvider in ./web3-config-provider for replace the one in @ant-design/web3-common
export { Web3ConfigProvider, type Web3ThemeConfig } from './web3-config-provider';
export * from './pay-panel';
