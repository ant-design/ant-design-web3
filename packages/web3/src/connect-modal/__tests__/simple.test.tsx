import { ConnectModal } from '@ant-design/web3';
import { fireEvent, render } from '@testing-library/react';
import { theme as antTheme, ConfigProvider } from 'antd';
import { describe, expect, it, vi } from 'vitest';

import { groupOrder, walletList } from './mock';

describe('ConnectModal without guide', () => {
  it.each(['light', 'dark'] as const)('should render in %s mode', async (theme) => {
    const App = () => (
      <ConfigProvider
        theme={{
          algorithm: theme === 'light' ? antTheme.defaultAlgorithm : antTheme.darkAlgorithm,
        }}
      >
        <ConnectModal
          open
          title="ConnectModal"
          footer="Powered by AntChain"
          group={{
            groupOrder,
          }}
          walletList={walletList}
        />
      </ConfigProvider>
    );
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
    // should have ant-web3-connect-modal class
    expect(baseElement.querySelector('.ant-modal')?.className).toContain('ant-web3-connect-modal');
    // should have simple class when without guide
    expect(baseElement.querySelector('.ant-web3-connect-modal-body')?.className).toContain(
      'simple',
    );

    // should have title and footer
    expect(baseElement.querySelector('.ant-web3-connect-modal-title')?.textContent).toBe(
      'ConnectModal',
    );
    expect(baseElement.querySelector('.ant-web3-connect-modal-footer')?.textContent).toBe(
      'Powered by AntChain',
    );

    // group order
    expect(baseElement.querySelectorAll('.ant-web3-connect-modal-group-title')[0].textContent).toBe(
      'Popular',
    );
    expect(baseElement.querySelectorAll('.ant-web3-connect-modal-group-title')[1].textContent).toBe(
      'More',
    );

    // wallet list items
    expect(baseElement.querySelectorAll('.ant-web3-connect-modal-wallet-item').length).toBe(
      walletList.length,
    );
  });
  it('render with mode simple', async () => {
    const App = () => (
      <ConnectModal open groupOrder={groupOrder} walletList={walletList} mode="simple" />
    );
    const { baseElement } = render(<App />);
    const btn = baseElement.querySelector('.ant-web3-connect-modal-simple-guide-right');
    fireEvent.click(btn!);
    await vi.waitFor(() => {
      expect(baseElement.querySelector('.ant-web3-connect-modal-simple-guide-panel')).not.toBe(
        null,
      );
    });
  });
});
