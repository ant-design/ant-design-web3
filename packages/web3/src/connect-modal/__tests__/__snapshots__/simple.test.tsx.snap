// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectModal without guide > should render in dark mode 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-1ikjzlw ant-web3-connect-modal-root"
    >
      <div
        class="ant-modal-mask ant-fade-appear ant-fade-appear-start ant-fade"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-1ikjzlw ant-web3-connect-modal css-dev-only-do-not-override-1ikjzlw ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 380px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-web3-connect-modal-body ant-web3-connect-modal-body-simple css-dev-only-do-not-override-1ikjzlw"
                >
                  <div
                    class="ant-web3-connect-modal-list-panel"
                  >
                    <div
                      class="ant-web3-connect-modal-header"
                    >
                      <h2
                        class="ant-web3-connect-modal-title"
                      >
                        ConnectModal
                      </h2>
                    </div>
                    <div
                      class="ant-web3-connect-modal-list"
                    >
                      <div
                        class="ant-web3-connect-modal-list-container"
                      >
                        <div
                          class="ant-web3-connect-modal-wallet-list"
                        >
                          <div
                            class="ant-web3-connect-modal-group"
                          >
                            <div
                              class="ant-web3-connect-modal-group-title"
                            >
                              Popular
                            </div>
                            <div
                              class="ant-web3-connect-modal-group-content"
                            >
                              <div
                                class="ant-list ant-list-split css-dev-only-do-not-override-1ikjzlw"
                              >
                                <div
                                  class="ant-spin-nested-loading css-dev-only-do-not-override-1ikjzlw"
                                >
                                  <div
                                    class="ant-spin-container"
                                  >
                                    <ul
                                      class="ant-list-items"
                                    >
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=0"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet3"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=3"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet3
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-web3-connect-modal-group"
                          >
                            <div
                              class="ant-web3-connect-modal-group-title"
                            >
                              More
                            </div>
                            <div
                              class="ant-web3-connect-modal-group-content"
                            >
                              <div
                                class="ant-list ant-list-split css-dev-only-do-not-override-1ikjzlw"
                              >
                                <div
                                  class="ant-spin-nested-loading css-dev-only-do-not-override-1ikjzlw"
                                >
                                  <div
                                    class="ant-spin-container"
                                  >
                                    <ul
                                      class="ant-list-items"
                                    >
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet2"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=1"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet2
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet4"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet4
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet5"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet5
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet6"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet6
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet6"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-1ikjzlw"
                                          >
                                            Test Wallet6
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-1ikjzlw ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ant-web3-connect-modal-footer-container"
                      >
                        <div
                          class="ant-web3-connect-modal-simple-guide"
                        >
                          New to crypto wallets?
                          <button
                            class="ant-btn css-dev-only-do-not-override-1ikjzlw ant-btn-link ant-btn-color-primary ant-btn-variant-link ant-btn-sm ant-web3-connect-modal-simple-guide-right"
                            type="button"
                          >
                            <span>
                              Learn More
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-web3-connect-modal-footer"
                        >
                          Powered by AntChain
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ConnectModal without guide > should render in light mode 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-ni1kz0 ant-web3-connect-modal-root"
    >
      <div
        class="ant-modal-mask ant-fade-appear ant-fade-appear-start ant-fade"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-ni1kz0 ant-web3-connect-modal css-dev-only-do-not-override-ni1kz0 ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 380px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-web3-connect-modal-body ant-web3-connect-modal-body-simple css-dev-only-do-not-override-ni1kz0"
                >
                  <div
                    class="ant-web3-connect-modal-list-panel"
                  >
                    <div
                      class="ant-web3-connect-modal-header"
                    >
                      <h2
                        class="ant-web3-connect-modal-title"
                      >
                        ConnectModal
                      </h2>
                    </div>
                    <div
                      class="ant-web3-connect-modal-list"
                    >
                      <div
                        class="ant-web3-connect-modal-list-container"
                      >
                        <div
                          class="ant-web3-connect-modal-wallet-list"
                        >
                          <div
                            class="ant-web3-connect-modal-group"
                          >
                            <div
                              class="ant-web3-connect-modal-group-title"
                            >
                              Popular
                            </div>
                            <div
                              class="ant-web3-connect-modal-group-content"
                            >
                              <div
                                class="ant-list ant-list-split css-dev-only-do-not-override-ni1kz0"
                              >
                                <div
                                  class="ant-spin-nested-loading css-dev-only-do-not-override-ni1kz0"
                                >
                                  <div
                                    class="ant-spin-container"
                                  >
                                    <ul
                                      class="ant-list-items"
                                    >
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=0"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet3"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=3"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet3
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-web3-connect-modal-group"
                          >
                            <div
                              class="ant-web3-connect-modal-group-title"
                            >
                              More
                            </div>
                            <div
                              class="ant-web3-connect-modal-group-content"
                            >
                              <div
                                class="ant-list ant-list-split css-dev-only-do-not-override-ni1kz0"
                              >
                                <div
                                  class="ant-spin-nested-loading css-dev-only-do-not-override-ni1kz0"
                                >
                                  <div
                                    class="ant-spin-container"
                                  >
                                    <ul
                                      class="ant-list-items"
                                    >
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet2"
                                            class="ant-web3-connect-modal-img"
                                            src="https://xsgames.co/randomusers/avatar.php?g=pixel&key=1"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet2
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet4"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet4
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet5"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet5
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet6"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet6
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                      <li
                                        class="ant-list-item ant-web3-connect-modal-wallet-item"
                                      >
                                        <div
                                          class="ant-web3-connect-modal-content"
                                        >
                                          <img
                                            alt="Test Wallet6"
                                            class="ant-web3-connect-modal-img"
                                          />
                                          <span
                                            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-single-line ant-web3-connect-modal-name css-dev-only-do-not-override-ni1kz0"
                                          >
                                            Test Wallet6
                                          </span>
                                        </div>
                                        <div
                                          class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                                        >
                                          <div
                                            class="ant-space-item"
                                          />
                                        </div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ant-web3-connect-modal-footer-container"
                      >
                        <div
                          class="ant-web3-connect-modal-simple-guide"
                        >
                          New to crypto wallets?
                          <button
                            class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-link ant-btn-color-primary ant-btn-variant-link ant-btn-sm ant-web3-connect-modal-simple-guide-right"
                            type="button"
                          >
                            <span>
                              Learn More
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-web3-connect-modal-footer"
                        >
                          Powered by AntChain
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
